base/inference.jl

```diff
@@ -3365,40 +3460,44 @@ function optimize(me::InferenceState)
     type_annotate!(me)
 
     # run optimization passes on fulltree
-    force_noinline = false
+    force_noinline = true
```

Comment:
It doesn't make sense to me that `force_noinline` would default to true. It's supposed to be true only for functions declared `@noinline`. In fact with this change I think it's now always true? Am I missing something?